src/GitHub.VisualStudio/Services/VSGitExtFactory.cs

```diff
@@ -13,12 +12,14 @@ public class VSGitExtFactory
         readonly int vsVersion;
         readonly IServiceProvider serviceProvider;
         readonly IGitService gitService;
+        readonly JoinableTaskContext joinableTaskContect;
 
-        public VSGitExtFactory(int vsVersion, IServiceProvider serviceProvider, IGitService gitService)
+        public VSGitExtFactory(int vsVersion, IServiceProvider serviceProvider, IGitService gitService, JoinableTaskContext joinableTaskContect)
```

Comment:
I noticed `[Import(AllowDefault = true)]` is not used here as well?

Reply:
In a similar way to `UsageTracker`, `VSGitExtFactory` is instantiated by a Visual Studio package and not MEF. `JoinableTaskContext` is passed from the package and will never be null.